### PR TITLE
Added 3D gimp functionality

### PR DESCRIPTION
**Pull request by cw646**
gimp_numbering

See 3D GIMP node numbering above

### CODE REVIEW

File: include/hexahedron_gimp_element.tcc [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        } else {
+          throw std::runtime_error(
+              "GIMP grad shapefn: Point location outside area of influence");
+        }
+      }
+      // see: Pruijn, N.S., 2016. Eq(4.32)
+      grad_shapefn(n, 0) = dni(0) * sni(1) * sni(2);
+      grad_shapefn(n, 1) = dni(1) * sni(0) * sni(2);
+      grad_shapefn(n, 2) = dni(2) * sni(1) * sni(0);
+    }
```
Comment (cw646) [inline include/hexahedron_gimp_element.tcc]: tweak ordering

File: include/hexahedron_gimp_element.tcc [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        } else {
+          throw std::runtime_error(
+              "GIMP shapefn: Point location outside area of influence");
+        }
+      }
+      shapefn(n) =
+          sni(0) * sni(1) * sni(2);  // See: Pruijn, N.S., 2016. Eq(4.30)
+    }
+  } catch (std::exception& exception) {
+    return shapefn;
```
Comment (kks32) [inline include/hexahedron_gimp_element.tcc]: Could you print out what the exception would be?
```
console_->error("{} # {}: {}", __FILE__, __LINE__, exception.what());
```

Comment (kks32) [reply include/hexahedron_gimp_element.tcc]: @cw646 could you do this please?

File: include/hexahedron_gimp_element.tcc (hunk 1 above)
Comment (kks32) [reply include/hexahedron_gimp_element.tcc]: And this?

**Review by kks32 — CHANGES_REQUESTED** [review]

**Review by kks32 — APPROVED** [review]

**Review by kks32 — APPROVED** [review]

**Pull request merged**